zio-http/src/test/scala/zio/http/middleware/MetricsSpec.scala

```diff
@@ -0,0 +1,86 @@
+package zio.http.middleware
+
+import zio.http.Middleware.metrics
+import zio.http._
+import zio.http.internal.HttpAppTestExtensions
+import zio._
+import zio.metrics.{Metric, MetricState}
+import zio.test._
+import zio.metrics.MetricLabel
+
+object MetricsSpec extends ZIOSpecDefault with HttpAppTestExtensions {
+  override def spec = suite("MetricsSpec")(
+    test("http_requests_total & http_errors_total") {
+      val app = Http
+        .collect[Request] {
+          case Method.GET -> !! / "ok"     => Http.ok
+          case Method.GET -> !! / "error"  => Http.error(HttpError.InternalServerError())
+          case Method.GET -> !! / "defect" => Http.die(new Throwable("boom"))
+        }
+        .flatten @@ metrics(extraLabels = Set(MetricLabel("test", "http_requests_total & http_errors_total")))
+
+      val total   = Metric.counterInt("http_requests_total").tagged("test", "http_requests_total & http_errors_total")
+      val totalOk = total.tagged("path", "/ok").tagged("method", "GET").tagged("status", "200")
+      val totalErrors   = total.tagged("path", "/error").tagged("method", "GET").tagged("status", "500")
+      val totalDefects  = total.tagged("path", "/defect").tagged("method", "GET").tagged("status", "500")
+      val totalNotFound = total.tagged("path", "/not-found").tagged("method", "GET").tagged("status", "404")
+
+      for {
+        _ <- app(Request(method = Method.GET, url = URL(!! / "ok")))
```

Comment:
Very minor - `Request.method` defaults to `GET`, so that bit can be removed from lines 30-32, 58, and 77.
However, you might prefer the more explicit approach here, since we're tagging things by the method.